src/rust/engine/process_execution/src/remote_cache.rs

```diff
@@ -453,7 +458,21 @@ impl crate::CommandRunner for CommandRunner {
             cached_response_opt
           }
           Err(err) => {
-            log::warn!("Failed to read from remote cache: {}", err);
+            let err_count = {
+              let mut errors_counter = self.read_errors_counter.lock();
+              let count = errors_counter.entry(err.clone()).or_insert(0);
+              *count += 1;
+              *count
+            };
+            let log_msg = format!(
+              "Failed to read from remote cache ({} occurrences so far): {}",
+              err_count, err
+            );
+            if err_count == 1 || err_count % 5 == 0 {
```

Comment:
Maybe pull the `5` out as a constant?

Reply:
+1. And include some of the explanation that is in the description for how it was chosen.